packages/ember-glimmer/tests/integration/helpers/element-action-test.js

```diff
@@ -0,0 +1,1340 @@
+    this.assert.equal(originalHandlerWasCalled, true, 'the click handler was called');
+  }
+
+  ['@test it should work properly in an #each block']() {
+    let editHandlerWasCalled = false;
+
+    let ExampleComponent = Component.extend({
+      items: emberA([1, 2, 3, 4]),
+      actions: {
+        edit() { editHandlerWasCalled = true; }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '{{#each items as |item|}}<a href="#" {{action "edit"}}>click me</a>{{/each}}'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      this.$('a').click();
+    });
+
+    this.assert.equal(editHandlerWasCalled, true, 'the event handler was called');
+  }
+
+  ['@test it should work properly in a {{#with foo as |bar|}} block']() {
+    let editHandlerWasCalled = false;
+
+    let ExampleComponent = Component.extend({
+      something: { ohai: 'there' },
+      actions: {
+        edit() { editHandlerWasCalled = true; }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '{{#with something as |somethingElse|}}<a href="#" {{action "edit"}}>click me</a>{{/with}}'
+    });
+
+    this.render('{{example-component}}');
+
+    this.runTask(() => {
+      this.$('a').click();
+    });
+
+    this.assert.equal(editHandlerWasCalled, true, 'the event handler was called');
+  }
+
+  ['@glimmer it should unregister event handlers on teardown, but not on rerender']() {
+    let editHandlerWasCalled = false;
+    let component;
+
+    let ExampleComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        component = this;
+      },
+      active: true,
+      actions: {
+        edit() { editHandlerWasCalled = true; }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '{{#if active}}<a href="#" {{action "edit"}}>click me</a>{{/if}}'
+    });
+
+    this.render('{{example-component}}');
+
+    let previousAttributes = getActionAttributes(component.$('a').get(0));
+
+    this.rerender();
+
+    let rerenderedAttributes = getActionAttributes(component.$('a').get(0));
+
+    this.runTask(() => {
+      component.set('active', false);
+    });
+
+    this.runTask(() => {
+      component.set('active', true);
+    });
+
+    let newAttributes = getActionAttributes(component.$('a').get(0));
+
+    this.assert.deepEqual(previousAttributes, rerenderedAttributes, 'the same action id is used across rerenders');
+    this.assert.notDeepEqual(previousAttributes, newAttributes, 'the action id is regenerated when the action is torn down');
+  }
+
+  ['@skip it should unregister event handlers on rerender']() {
+    // TODO: Port this test for HTMLBars only.
+  }
+
+  ['@skip it should unregister event handlers on inside virtual views']() {
```

Comment:
Agreed. The existing test does this by probing the `ActionManager` which feels wrong. I'm not sure any other way to observe the cleanup without inspecting internals.